In [1]:
from pathlib import Path
from abstract_search_evaluation import get_scores

/home/adamp/school/cmpt419/layman-scientific-embeddings/venv/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
MODELS_PATH = Path("models")

In [3]:
def print_model_scores(model_path: str | Path):
    scores = get_scores(model_path)
    print(model_path)
    print(f"Full match score: {scores["perfect_match_score"] * 100:.2f}%")
    print(f"In top results score: {scores["in_top_results_score"] * 100:.2f}%")

In [4]:
for model_path in MODELS_PATH.iterdir():
    print_model_scores(model_path)
    print("")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3456.30it/s]


models/vanilla-scibert
Full match score: 6.03%
In top results score: 27.57%



Loading weights: 100%|██████████| 310/310 [00:00<00:00, 3847.68it/s]
Loading weights: 0it [00:00, ?it/s]
Qwen3Model LOAD REPORT from: models/vanilla-qwen
Key                                                                     | Status     | 
------------------------------------------------------------------------+------------+-
base_model.model.layers.{0...27}.mlp.down_proj.lora_A.default.weight    | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.up_proj.lora_A.default.weight      | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.q_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.o_proj.lora_A.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.v_proj.lora_B.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.self_attn.k_proj.lora_A.default.weight | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.up_proj.lora_B.default.weight      | UNEXPECTED | 
base_model.model.layers.{0...27}.mlp.gate_proj.lora_A.

models/vanilla-qwen
Full match score: 6.03%
In top results score: 23.42%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3544.08it/s]


models/jargon-layman-scibert-0.75
Full match score: 6.13%
In top results score: 28.95%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3550.02it/s]


models/jargon-layman-scibert-1.00
Full match score: 6.23%
In top results score: 30.14%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3526.04it/s]


models/jargon-layman-scibert-0.25
Full match score: 6.23%
In top results score: 28.75%



Loading weights: 100%|██████████| 199/199 [00:00<00:00, 3363.23it/s]


models/jargon-layman-scibert-0.50
Full match score: 5.83%
In top results score: 28.56%



In [5]:
print_model_scores("allenai/scibert_scivocab_uncased")

Loading weights: 100%|██████████| 199/199 [00:00<00:00, 18660.10it/s]
BertModel LOAD REPORT from: allenai/scibert_scivocab_uncased
Key                                        | Status     |  | 
-------------------------------------------+------------+--+-
cls.predictions.transform.LayerNorm.weight | UNEXPECTED |  | 
cls.predictions.transform.dense.weight     | UNEXPECTED |  | 
cls.predictions.decoder.bias               | UNEXPECTED |  | 
cls.seq_relationship.bias                  | UNEXPECTED |  | 
cls.predictions.transform.LayerNorm.bias   | UNEXPECTED |  | 
cls.seq_relationship.weight                | UNEXPECTED |  | 
cls.predictions.decoder.weight             | UNEXPECTED |  | 
cls.predictions.bias                       | UNEXPECTED |  | 
cls.predictions.transform.dense.bias       | UNEXPECTED |  | 

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.


allenai/scibert_scivocab_uncased
Full match score: 1.48%
In top results score: 6.32%


# Results

| Model | Exact abstract match score | Abstract in top 5 results score |
|---|---|---|
| allenai/scibert_scivocab_uncased | 0.89% | 4.45% |
| vanilla-scibert | <u>6.03%</u> | <u>27.57%</u> |
| jargon-layman-scibert | **6.23%** | **30.14%** |
| vanilla-qwen | <u>6.03%</u> | 23.42% |